demos/cairo/cairo_clock/clock.d

```diff
@@ -35,12 +36,14 @@ public:
 	this()
 	{
 		//Attach our expose callback, which will draw the window.
-		addOnDraw(&drawCallback);
+		setDrawFunc ((area, cairo, w, h, data) {
+            (cast(Clock)data).drawCallback (new Context (cairo));
+        }, cast(void *)this, null);
```

Comment:
missing D wrapper of setDrawFunc to transfer context scope? (could access `this` in callback immediately then and not do the unsafe void* casts to this)

Should also probably not call GC (`new Context`) inside a draw function and rather use something like `scope c = new Context(cairo);` and `scope Context cr` as parameter to use stack allocated Context class?

Reply:
I think that it is possible to write more comfortable wrapper for setDrawFunc(), but there is a small question.
Is it better to create a something like public draw(Scoped!Canvas) method for the DrawingArea class and allow to override it or use addOnDraw(delegate (Scoped!Context))?
And I agree that using Scoped~Context is better. Will be replaced